go/libraries/doltcore/sqle/dsess/globalstate.go

```diff
@@ -40,35 +41,61 @@ func NewGlobalStateStoreForDb(ctx context.Context, dbName string, db *doltdb.Dol
 	rootRefs = append(rootRefs, branches...)
 	rootRefs = append(rootRefs, remotes...)
 
-	var roots []doltdb.Rootish
+	rootRefsChan := make(chan doltdb.Rootish, len(rootRefs))
```

Comment:
Instead of a channel here, just allocate `roots` here:

```go
roots := make([]doltdb.Rootish, len(rootRefs))
```

and then have the loop capture the iteration variable and write the root into the right place:

```go
for i, b := range rootRefs {
    eg.Go(func() error {
        ...
            roots[i] = cm
        ...
            roots[i] = ws
        ...
    })
}
```

Probably best to `eg.SetLimit()` on this errgroup as well.